File: src/test/console/java/src/main/java/web/NestedFramesPage.java
package web;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;

import java.util.concurrent.TimeUnit;

public class NestedFramesPage extends Base {
  @FindBy (xpath = "//frame[@name=\"frame-left\"]")
  public WebElement frameLeft;
  @FindBy (xpath = "//frame[@name=\"frame-right\"]")
  public WebElement frameRight;
  @FindBy (xpath = "//frame[@name=\"frame-middle\"]")
  public WebElement frameMiddle;
  @FindBy (xpath = "//frame[@name=\"frame-bottom\"]")
  public WebElement frameBottom;

  public NestedFramesPage(WebDriver driver, long timeoutInSeconds){
    super(driver, timeoutInSeconds, "http://the-internet.herokuapp.com/nested_frames");
  }

  public void open() throws Exception {
    super.open();
    this.getDriver().manage().timeouts().pageLoadTimeout(10, TimeUnit.SECONDS);
  }

  public void switchToFrameLeft() throws Exception {
    this.waitForFrameVisible(frameLeft);
    this.getDriver().switchTo().frame(frameLeft);
  }

  public void switchToFrameMiddle() throws Exception {
    this.waitForFrameVisible(frameMiddle);
    this.getDriver().switchTo().frame(frameMiddle);
  }

  public void switchToFrameRight() throws Exception {
    this.waitForFrameVisible(frameRight);
    this.getDriver().switchTo().frame(frameRight);
  }

  public void switchToFrameBottom() throws Exception {
    this.waitForFrameVisible(frameBottom);
    this.getDriver().switchTo().frame(frameBottom);
  }

  public void switchToParentFrame() throws Exception {
    this.getDriver().switchTo().parentFrame();
  }
}

File: src/test/console/dotnet/Jsonwired/web/NestedFramesPage.cs
using System;
using OpenQA.Selenium;
using OpenQA.Selenium.Support.PageObjects;

namespace web {
	public class NestedFramesPage : Base {
	  // Page elements
		String frameLeft { get; } = "frame-left";
		String frameRight { get; } = "frame-right";
		String frameMiddle { get; } = "frame-middle";
		String frameBottom { get; } = "frame-bottom";

		public NestedFramesPage(IWebDriver driver, long timeoutInSeconds):
			base(driver, timeoutInSeconds, "http://the-internet.herokuapp.com/nested_frames")	{
			}

		public override void open() {
			base.open();
			this.waitForFrameVisible(frameLeft);
		}

	  public void switchToFrameLeft() {
			this.getDriver().SwitchTo().Frame(frameLeft);
		}

		public void switchToFrameMiddle() {
			this.getDriver().SwitchTo().Frame(frameMiddle);
		}

		public void switchToFrameRight() {
			this.getDriver().SwitchTo().Frame(frameRight);
		}

		public void switchToFrameBottom() {
			this.getDriver().SwitchTo().Frame(frameBottom);
		}

		public void switchToParentFrame() {
	    	this.getDriver().SwitchTo().ParentFrame();
		}
	}
}
